src/http/response.rs

```diff
@@ -148,3 +127,25 @@ impl AsyncRead for IncomingBody {
         Ok(len)
     }
 }
+
+impl IncomingBody {
+    pub async fn read_all(&mut self) -> crate::io::Result<Vec<u8>> {
+        let mut buf = Vec::with_capacity(self.content_length.unwrap_or(CHUNK_SIZE) as usize);
+
+        loop {
+            // Wait for an event to be ready
+            let pollable = self.body_stream.subscribe();
+            self.reactor.wait_for(pollable).await;
+
+            match self.body_stream.read(CHUNK_SIZE) {
+                Ok(mut bytes) => buf.append(&mut bytes),
+                Err(StreamError::LastOperationFailed(err)) => {
+                    return Err(std::io::Error::other(err.to_debug_string()));
+                }
+                Err(StreamError::Closed) => break,
+            };
+        }
+
+        Ok(buf)
+    }
```

Comment:
Ohh yeah, this is neat! - Reading the entire body into a vec is a common pattern.

Fhis would probably be a good fit to generalize as part of `wstd::io::Read` trait as an async dual to [`std::io::Read::read_to_end`](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_end). Could you add this there instead? That way all other types can reuse this functionality, as long as they implement the `Read` trait.